gcloud-java-storage/src/main/java/com/google/gcloud/spi/DefaultStorageRpc.java

```diff
@@ -465,15 +465,18 @@ public void write(String uploadId, byte[] toWrite, int toWriteOffset, long destO
       GenericUrl url = new GenericUrl(uploadId);
       HttpRequest httpRequest = storage.getRequestFactory().buildPutRequest(url,
           new ByteArrayContent(null, toWrite, toWriteOffset, length));
-      long limit = destOffset + length;
-      StringBuilder range = new StringBuilder("bytes ");
-      range.append(destOffset).append('-').append(limit - 1).append('/');
-      if (last) {
-        range.append(limit);
-      } else {
-        range.append('*');
+      // 0-length requests are valid, don't need content-range.
```

Comment:
Actually, according to @mziccard that would only work if nothing was already written to GCS. We are discussing the issue with the GCS people (a clean solution is unknown yet).
One workaround is to make BlobWriteChannel (or BaseWriteChannel as I imagine same issue applies to bigquery `TableDataWriteChannel`?) keep/remember the last block and use it (with
the correct offset) for zero bytes (and final is true - make it no-op otherwise).

@jean-philippe-martin you can keep your change as is though maybe add a comment to remove once #720 is fixed.